bin/rubocop

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'bundler/inline'
+
+gemfile do
+  source 'https://rubygems.org'
+
+  gem 'rubocop', '=  1.66.1'
+end
+
+# Copied from `exe/rubocop` from the Rubocop gem
```

Comment:
I think you should be able to use `require Gem.bin_path('rubocop', 'rubocop')` instead to avoid having to copy the logic from `exe/rubocop`.

Reply:
This did result in a LoadError, even though the file does exist (probably because it expects a `.rb` extension). Using `Kernel.exec` works.

Reply:
Right, `require` is picky about the extension.
`load` would likely work then.

I think `exec` might lose the state from bundler above and potentially run a different version of RuboCop, so could you try `load`?

Alternatively we could exec but pass the version too like:
`exec(Gem.bin_path('rubocop', 'rubocop'), '_1.66.1_', *ARGV)`
but that probably won't work if bundle `path` config is set (which I have).

Reply:
The bin_path includes the full version in the path (so in my case it's something like `~/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/exe/rubocop`). I've tried it with switching to a different version and running `bin/rubocop -v`, and this shows the correct version.

Reply:
You're right it seems to be correct:
```
$ cat /home/eregon/.rubies/ruby-3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.56.3/exe/rubocop 
#!/usr/bin/env ruby
# frozen_string_literal: true

$LOAD_PATH.unshift("#{__dir__}/../lib")
...
```
because of that $LOAD_PATH modification.
Gems are not supposed to do that but whatever, it helps here.

The ultra correct way would be to replicate what `bin/rubocop` does:
```ruby
cat ~/.rubies/ruby-3.2.2/bin/rubocop 
#!/home/eregon/.rubies/ruby-3.2.2/bin/ruby
#
# This file was generated by RubyGems.
#
# The application 'rubocop' is installed as part of a gem, and
# this file is here to facilitate running it.
#

require 'rubygems'

Gem.use_gemdeps

version = ">= 0.a"

str = ARGV.first
if str
  str = str.b[/\A_(.*)_\z/, 1]
  if str and Gem::Version.correct?(str)
    version = str
    ARGV.shift
  end
end

if Gem.respond_to?(:activate_bin_path)
load Gem.activate_bin_path('rubocop', 'rubocop', version)
else
gem "rubocop", version
load Gem.bin_path("rubocop", "rubocop", version)
end
```
Which explicitly activates the gem with the optionally-given version (and otherwise picks latest).

Anyway, the exec seems fine for now, we can always improve this later if needed.